### PR TITLE
[merged] Use git.mk

### PR DESCRIPTION
**Pull request by cgwalters**
It's a lot nicer than manually maintaining .gitignore in general.

### CODE REVIEW

File: Makefile-tests.am [hunk 1]
```diff
@@ -223,6 +223,7 @@ EXTRA_DIST += \
 tests-libreaddir-rand-so-symlink:
 	ln -fns ../.libs/libreaddir-rand.so tests
 ALL_LOCAL_RULES += tests-libreaddir-rand-so-symlink
+CLEANFILES += tests/libreaddir-rand.so
```
Comment (cgwalters) [inline Makefile-tests.am]: Worth noting changes like this one are now required.

**Comment by giuseppe** [comment]
I got an error when I tried to `rm .gitignore && make gitignore`.  This patch fixes it for me:

```
diff --git a/Makefile-decls.am b/Makefile-decls.am
index eefffd6..d8ec5ab 100644
--- a/Makefile-decls.am
+++ b/Makefile-decls.am
@@ -43,8 +43,6 @@ gir_DATA =
 typelibdir = $(libdir)/girepository-1.0
 typelib_DATA =
 gsettings_SCHEMAS =
-# git.mk
-GITIGNOREFILES =
 
 # This is a special facility to chain together hooks easily
 INSTALL_DATA_HOOKS =
diff --git a/Makefile.am b/Makefile.am
index 488d4b6..6c4d18e 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -20,10 +20,10 @@ include Makefile-decls.am
 shortened_sysconfdir = $$(echo "$(sysconfdir)" | sed -e 's|^$(prefix)||' -e 's|^/||')
 
 ACLOCAL_AMFLAGS = -I buildutil ${ACLOCAL_FLAGS}
-AM_CPPFLAGS += -DDATADIR='"$(datadir)"' -DLIBEXECDIR='"$(libexecdir)"' \
+AM_CPPFLAGS += -DDATADIR="$(datadir)" -DLIBEXECDIR="$(libexecdir)" \
        -DLOCALEDIR=\"$(datadir)/locale\" -DSYSCONFDIR=\"$(sysconfdir)\" \
        -DSHORTENED_SYSCONFDIR=\"$(shortened_sysconfdir)\" \
-       -DOSTREE_FEATURES='"$(OSTREE_FEATURES)"' \
+       -DOSTREE_FEATURES='$(OSTREE_FEATURES)' \
        -DG_LOG_DOMAIN=\"OSTree\" \
        -DGLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_40 -DGLIB_VERSION_MAX_ALLOWED=GLIB_VERSION_2_40 \
        -DSOUP_VERSION_MIN_REQUIRED=SOUP_VERSION_2_40 -DSOUP_VERSION_MAX_ALLOWED=SOUP_VERSION_2_48
```

**Comment by cgwalters** [comment]
I've imported the first hunk into a fixup, but I don't understand how the second hunk is related to this.  If we're going to do this we should be consistent and use `\"`, not `'` I'd say, right?

But what's the error on your end?

**Comment by giuseppe** [comment]
ah true, `'` won't be enough as it causes a build error, I didn't try to `make clean` before.

If I leave `"VALUE"` as in the original version, I get this error:
```
$ rm .gitignore && make
git.mk: Generating .gitignore
/bin/sh: line 111: test: too many arguments
```

**Comment by cgwalters** [comment]
I spent too long debugging this but anyways it turned out to be https://github.com/behdad/git.mk/pull/32

**Comment by cgwalters** [comment]
Rebased :surfer:

**Comment by giuseppe** [comment]
@cgwalters-bot  r+ cf24290

**Comment by cgwalters** [comment]
@cgwalters-bot retry

**Pull request closed without merging**